Add maxAuthLifetime configuration to SamlConfig

mikelupo opened the pull request:
maxAuthLifetime = 86400
If no entry included in the gerrit.config the default of
86400 (24 * 60 * 60) is assumed.

This was recently added to the SAML plugin for jenkins to combat
org.pac4j.saml.exceptions.SAMLException:
   Authentication issue instant is too old or in the future
Likewise, this should help with gerrit.

Logs in error_log:
[main] INFO  com.thesamet.gerrit.plugins.saml.SamlWebFilter
    : Max Authentication Lifetime: 86400

Builds against:
 - gerrit-plugin-api version 2.13.6
 - org.pac4j.pac4j-saml version 2.0.0-RC1

Forced change from pac4j-saml update removes RequiresHttpAction
in favor of HTTPAction. Also removes boolean parameter
from saml2Client.redirect(context, true) to saml2Client.redirect(context);

akamlupo committed:
Add maxAuthLifetime configuration to SamlConfig

maxAuthLifetime = 86400
If no entry included in the gerrit.config the default of
86400 (24 * 60 * 60) is assumed.

This was recently added to the SAML plugin for jenkins to combat
org.pac4j.saml.exceptions.SAMLException:
   Authentication issue instant is too old or in the future
Likewise, this should help with gerrit.

Logs in error_log:
[main] INFO  com.thesamet.gerrit.plugins.saml.SamlWebFilter
    : Max Authentication Lifetime: 86400

Builds against:
 - gerrit-plugin-api version 2.13.6
 - org.pac4j.pac4j-saml version 2.0.0-RC1

Forced change from pac4j-saml update removes RequiresHttpAction
in favor of HTTPAction. Also removes boolean parameter
from saml2Client.redirect(context, true) to saml2Client.redirect(context);

mikelupo commented:
@thesamet  I see an omission from the doc in regards to the newer version of the .jar file. I will fix this when I incorporate changes based on your anticipated comments.
I don't know how to include a test for this and could use some guidance in this regard. I'm a SAML newb. Thanks!

mikelupo commented:
Hi @thesamet Curious if you can review this change. Thanks!

mikelupo commented:
Hi @thesamet Curious if you can review this change. Thanks!

thesamet commented on src/main/java/com/thesamet/gerrit/plugins/saml/SamlWebFilter.java:
```diff
@@ -67,10 +67,13 @@ private String getHeaderFromConfig(Config gerritConfig, String name) {
     @Inject
     SamlWebFilter(@GerritServerConfig Config gerritConfig, SamlConfig samlConfig) {
         this.samlConfig = samlConfig;
+        log.info("Max Authentication Lifetime: "+ samlConfig.getMaxAuthLifetime());
```
log.debug?

thesamet replied:
space before and after +

thesamet commented on src/main/java/com/thesamet/gerrit/plugins/saml/SamlConfig.java:
```diff
@@ -74,6 +78,19 @@ public String getEmailAddressAttr() {
+    try {
+      maxLifetime = Integer.parseInt(getMaxAuthLifetimeAttr());
+    } catch (NumberFormatException nfe) {
+      SamlWebFilter.logError("Error reading \"maxAuthLifetime\" attribute in gerrit.config. Please use digits only");
```
Please add a logger for this class instead of using the logger in SamlWebFilter.

thesamet commented on src/main/java/com/thesamet/gerrit/plugins/saml/SamlWebFilter.java:
```diff
@@ -264,6 +267,14 @@ public String getHeader(String name) {
         }
     }
 
+    public static void logWarning(String message) {
```
please remove these two functions.

thesamet commented on src/main/java/com/thesamet/gerrit/plugins/saml/SamlConfig.java:
```diff
@@ -32,6 +32,8 @@
   private final String displayNameAttr;
   private final String userNameAttr;
   private final String emailAddressAttr;
+  private final String maxAuthLifetimeAttr;
```
why not Int?

thesamet commented on src/main/java/com/thesamet/gerrit/plugins/saml/SamlConfig.java:
```diff
@@ -74,6 +78,19 @@ public String getEmailAddressAttr() {
     return emailAddressAttr;
   }
 
+  public String getMaxAuthLifetimeAttr() { return maxAuthLifetimeAttr; }
```
should be a single function that returns int.

thesamet commented on src/main/java/com/thesamet/gerrit/plugins/saml/SamlWebFilter.java:
```diff
@@ -159,21 +162,21 @@ public void doFilter(ServletRequest incomingRequest, ServletResponse response,
         context.setSessionAttribute(SAML2Client.SAML_RELAY_STATE_ATTRIBUTE, redirectUri);
         log.debug("Setting redirectUri: {}", redirectUri);
-        saml2Client.redirect(context, true);
+        saml2Client.redirect(context);
```
can you elaborate on this change - why is it needed?

mikelupo replied:
Because the newer Pac4J changed the API to only one argument. The context.

thesamet commented:
Looks good overall, a few minor comments.

akamlupo committed:
Update Code based on comments by @thesamet

 - Change method getting  SAML maxAuthLifetime from string
method to int method.
 - Remove logging methods that are unneeded.
